network/p2p/libp2pNode.go

```diff
@@ -490,28 +513,59 @@ func (n *Node) IsConnected(identity flow.Identity) (bool, error) {
 	return isConnected, nil
 }
 
-// bootstrapLibP2PHost creates and starts a libp2p host as well as a pubsub component for it, and returns all in a
-// libP2PHostWrapper.
-// In case `allowList` is true, it also creates and embeds a connection gater in the returned libP2PHostWrapper, which
-// whitelists the `allowListAddres` nodes.
-func bootstrapLibP2PHost(ctx context.Context,
-	logger zerolog.Logger,
-	address string,
-	conMgr *ConnManager,
-	key crypto.PrivKey,
-	allowList bool,
-	psOption ...pubsub.Option) (host.Host, *connGater, *pubsub.PubSub, error) {
+// LibP2PHost returns a libp2p host initialized to listen on the given address and using the given private key and
+// customized with options
+func LibP2PHost(ctx context.Context, address string, key fcrypto.PrivateKey, options ...config.Option) (host.Host, error) {
+
+	defaultOptions, err := DefaultLibP2POptions(address, key)
+	if err != nil {
+		return nil, err
+	}
+
+	allOptions := append(defaultOptions, options...)
+
+	// create the libp2p host
+	libP2PHost, err := libp2p.New(ctx, allOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("could not create libp2p host: %w", err)
+	}
+
+	return libP2PHost, nil
 
-	var connGater *connGater
+}
+
+func WithLibP2PPingEnabled() config.Option {
+	return libp2p.Ping(true)
+}
+
+func WithLibP2PPingDisabled() config.Option {
+	return libp2p.Ping(true)
+}
```

Comment:
I think we can take a boolean as argument of `WithLibP2PPing`, pass it to `libp2p.Ping`, and correct the issue you have in `..Disabled` here.